docs/building.md

```diff
@@ -199,3 +199,173 @@ added to the CMAKE_MODULE_PATH.
 ~~~~~~~~~~~~~{.sh}
 cmake -DBUILDSWIG=OFF -DBUILDARCH=PERIPHERALMAN -DANDROID_TOOLCHAIN_NAME=x86-i686 -DCMAKE_TOOLCHAIN_FILE=/path/to/android-ndk-r14b/build/cmake/android.toolchain.cmake -DCMAKE_MODULE_PATH=/path/to/native-libandroidthings ..
 ~~~~~~~~~~~~~
+
+## Building with Docker
+
+You can use `docker` and `docker-compose` to generate a complete build environment
+for mraa without having to install any other tool.
+
+Requirements:
+* [docker](https://www.docker.com/get-docker) >= 1.12.6
+* [docker-compose](https://docs.docker.com/compose/install/) >= 1.9.0
+
+**NOTE:** docker-compose is an optional requirement. It actually make running complex
+docker build and run command easier. But you can just use docker to build and run.
+
+### Docker Images Hierarchy
+
+To improve build times and images sizes, mraa use a build hierarchy to incrementally
+create a build environment. Find below a brief description of them:
+
+1. **mraa-base:** Provides the basic infrastructure and tools to compile C/C++ code and documentation.
+2. **mraa-python:** Provides the python2/python3 build tools. Depends on `mraa-base`.
+3. **mraa-java:** Provides the Java build tools. Depends on `mraa-base`.
+4. **mraa-android:** Provides the Android Things build tools.  Depends on `mraa-java`.
+5. **mraa-node4:** Provides the Node.js v4.4.7 build tools. Depends on `mraa-base`.
+6. **mraa-node5:** Same as `mraa-node4`, but using Node.js v5.12.0.
```

Comment:
Thanks for adding that, in general the change looks ok to me by inspection, but I also plan to do some more detailed tests later this week and will post final review after that.